spec/acceptance/basic_candlepin_spec.rb

```diff
@@ -9,7 +9,25 @@
     it { is_expected.to be_listening }
   end
 
-  describe command('curl -k -s -o /dev/null -w \'%{http_code}\' https://localhost:8443/candlepin/status') do
-      its(:stdout) { should eq "200" }
+  describe command("curl -k -s -o /dev/null -w '%{http_code}' https://localhost:8443/candlepin/status") do
+    its(:stdout) { should eq "200" }
+  end
+
+  describe command("curl -s -o /dev/null -w '%{http_code}' --key /etc/candlepin/certs/client-cert.key --cert /etc/candlepin/certs/client-cert.pem --cacert /etc/candlepin/certs/candlepin-ca.crt --tlsv1.3 --tls-max 1.3 https://#{fact('fqdn')}:8443/candlepin/status/") do
```

Comment:
I would recommend to store the output of `fact('fqdn')` somewhere. Every time you use it, some SSH connection is used to call facter and there is no caching. Other than the overhead, this also creates a lot of extra log lines that obfuscate the important parts. Perhaps even some `base_curl_command` variable to avoid a lot of the repetition.

Reply:
Oh, and https://github.com/theforeman/puppet-pulpcore/blob/master/spec/support/acceptance/curl_command.rb is probably also useful here. It gives a much cleaner API like https://github.com/theforeman/puppet-pulpcore/blob/5462f346da26df650290e9ab19cb6a522feb8632/spec/acceptance/basic_spec.rb#L73-L76. It would need some additional code for the TLS options, but that shouldn't be too complicated.